### ISSUE TITLE
az vm create from an image without a configured vmImageAliasDoc fails to create the vm

### ISSUE THREAD

**Issue by carlowahlstedt** [Azure/azure-cli#12902]
## Describe the bug
After registering a new cloud (`az cloud register`), if attempt to create a vm from an image without first setting `--endpoint-vm-image-alias-doc` to some value then you cannot create the vm.

**Command Name**
`az vm create`

**Errors:**
```
'endpoint_vm_image_alias_doc' isn't configured. Please invoke 'az cloud update' to configure it or use '--all' to retrieve images from server
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Register a new cloud `az cloud register -n NewAzureCloud --endpoint-resource-manager "https://management.azure.com/" --endpoint-management "https://management.core.windows.net/"`
- Run the command to confirm the `vmImageAliasDoc` is set to null (`az cloud show -n NewAzureCloud -o json`)
- Run `az login` to authenticate to the cloud
- Create an image within a resource group that you will also create the vm
- With the created image, run `az vm create -g [your group name] -n [your vm name] --image [your image name] -l [location] --admin-username [admin name] --subnet [I happened to use a subnet because I had one accessible] --generate-ssh-keys`

## Expected Behavior
The vm should be created from the image.

## Environment Summary
```
macOS-10.15.4-x86_64-i386-64bit
Python 3.8.2
Installer: HOMEBREW

azure-cli 2.3.1

Extensions:
azure-devops 0.16.0
image-copy-extension 0.2.3

```
## Additional Context
The error happens [here]([https://github.com/Azure/azure-cli/blob/bdaf8dc5eb7eaa218fae6b785d8baffaae655cf1/src/azure-cli/azure/cli/command_modules/vm/_actions.py#L95), because the endpoint is null. It is called from [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/_image_builder.py#L252) as the fourth check. If the check didn't throw an exception then it would continue to step #5 and succeed.

You can verify this by:
- running `az cloud update -n NewAzureCloud --endpoint-vm-image-alias-doc "https://im.an.endpoint"`
- rerunning your `az vm create` statement from above

This time you will see `Failed to retrieve image alias doc 'https://im.an.endpoint'. Error: 'ConnectionError'. Use local copy instead.` and then it will continue to create the vm. The [local copy it references](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/_alias.py) is a hardcode of the [default setting](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-compute/quickstart-templates/aliases.json) for the cloud AzureCloud (`az cloud show -n AzureCloud -o json`).

If you'd like a PR on this, then I'm willing to try if there's direction on how you prefer it fixed. Easiest solution seems to not make a null endpoint throw an exception, but that likely has other repercussions I'm unaware of.

**Comment by yonzhan** [comment]
@qwordy  please take a look

**Comment by qwordy** [comment]
Thank you for finding this bug. I write a PR to fix it. You analysis for the cause helps me a lot.
https://github.com/Azure/azure-cli/pull/13022

**Comment by qwordy** [comment]
PR is merged.

**Comment by qwordy** [comment]
It is fixed in version 2.5.1.
